libs/dissect/src/main/java/org/opensearch/dissect/DissectParser.java

```diff
@@ -284,7 +284,11 @@ public Map<String, String> parse(String inputString) {
                         delimiter = dissectPair.getDelimiter().getBytes(StandardCharsets.UTF_8);
                         // i is always one byte after the last found delimiter, aka the start of the next value
                         valueStart = i;
+                    } else {
```

Comment:
Any chance `i` doesn't change above? I would at least assert or something, after all we *are* fixing an infinite loop here.

Reply:
Seems not, I've added an assertion for this.